### PR TITLE
Add list of allowed domains for Watcher email action 

### LINKED ISSUE #84739: Add configurable list of allowed domains for Watcher email action
### Description

When using the Watcher email action, it would be beneficial to be able to specify a list of allowed domains to which emails are allowed to be sent. This would probably be a non-dynamic setting, and allow specifying multiple domains.

### PR DESCRIPTION
**Pull request by dakrone**
This adds the `xpack.notification.email.account.domain_allowlist` dynamic cluster setting that
allows an administrator to specify a list of domains to which emails are allowed to be sent. The
default value for this setting is `["*"]` which means all domains are allowed. It supports
rudimentary globbing (`*`) in the domain name, so `*.company.com` will work as a valid option.

Resolves https://github.com/elastic/elasticsearch/issues/84739

### CODE REVIEW

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by dakrone** [comment]
@elasticmachine update branch

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/EmailService.java [hunk 1]
```diff
@@ -174,10 +192,16 @@ public EmailService(Settings settings, @Nullable CryptoService cryptoService, SS
         clusterSettings.addAffixUpdateConsumer(SETTING_SMTP_LOCAL_PORT, (s, o) -> {}, (s, o) -> {});
         clusterSettings.addAffixUpdateConsumer(SETTING_SMTP_SEND_PARTIAL, (s, o) -> {}, (s, o) -> {});
         clusterSettings.addAffixUpdateConsumer(SETTING_SMTP_WAIT_ON_QUIT, (s, o) -> {}, (s, o) -> {});
+        this.allowedDomains = new HashSet<>(SETTING_DOMAIN_ALLOWLIST.get(settings));
+        clusterSettings.addSettingsUpdateConsumer(SETTING_DOMAIN_ALLOWLIST, (s) -> {});
```
Comment (jbaiera) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/EmailService.java:196]: Question for my own curiosity: What is the benefit of this consumer if it doesn't perform any actions? It looks like we mark the setting as dynamic, but we don't actually update it anywhere?

**Review by jbaiera — APPROVED** [review]
LGTM, question aside

**Comment by Kushmaro** [comment]
Will this PR also affect Kibana Alerting's SMTP connectors?

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/EmailService.java (hunk 1 above)
Comment (dakrone) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/EmailService.java:196]: According to a comment above, it ensures that the setting is logged when it's changed, but I didn't test to ensure that (just followed the other examples)

Comment (masseyke) [reply x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/EmailService.java:196]: I think it needs to update `allowedDomains` -- https://github.com/elastic/elasticsearch/pull/90426

**Pull request merged**